### DISCUSSION TITLE
Paddle2.0不支持 Intel 奔腾双核 E5700等老旧处理器

### DISCUSSION THREAD

**Discussion by kjf4096** [PaddlePaddle/PaddleNLP#687] [Q&A] [answered]
Paddle1.x支持 Intel 奔腾双核 E5700等老旧处理器，但升级到2.0以后就不支持了。
发现老电脑可以安装2.0版本的paddle，但是import paddle就会出错，显示如下：
$ python
Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import paddle
WARNING: AVX is not support on your machine. Hence, no_avx core will be imported, It has much worse preformance than avx core.
Error: Can not import noavx core while this file exists: D:\Program Files\python\lib\site-packages\paddle\fluid\core_noavx.pyd
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "D:\Program Files\python\lib\site-packages\paddle\__init__.py", line 29, in <module>
    from .fluid import monkey_patch_variable
  File "D:\Program Files\python\lib\site-packages\paddle\fluid\__init__.py", line 35, in <module>
    from . import framework
  File "D:\Program Files\python\lib\site-packages\paddle\fluid\framework.py", line 36, in <module>
    from . import core
  File "D:\Program Files\python\lib\site-packages\paddle\fluid\core.py", line 349, in <module>
    raise e
  File "D:\Program Files\python\lib\site-packages\paddle\fluid\core.py", line 328, in <module>
    from .core_noavx import _cuda_synchronize
ImportError: cannot import name '_cuda_synchronize' from 'paddle.fluid.core_noavx' (D:\Program Files\python\lib\site-packages\paddle\fluid\core_noavx.pyd)
现在很多人都还在用10年前的电脑，建议paddlepaddle能够兼容一下老电脑，使更多人能够使用paddlepaddle。

**Comment by wawltor** [comment]
麻烦提供一下具体的PaddlePaddle的版本，目前最新的PaddlePaddle的版本已经是2.1.1，我们需要确定一下PaddlePaddle的版本来反馈一下问题

**Comment by kjf4096** [comment]
从2.0.0开始的所有版本，import paddle都会出错

**Comment by wawltor** [comment]
> 从2.0.0开始的所有版本，import paddle都会出错

收到 我反馈一下问题

**Comment by wawltor** [comment, marked as answer]
https://www.paddlepaddle.org.cn/documentation/docs/zh/install/pip/windows-pip.html  @kjf4096  可以尝试安装一下no_avx的包
<img width="966" alt="355a158c23a048cedaff8eed53e137a9" src="https://user-images.githubusercontent.com/16698950/124093122-21a90880-da8a-11eb-8a00-a0f8205e410a.png">

**Comment by kjf4096** [comment]
OK 安装成功了。
D:\Program Files\cmder_mini
$ pip install decorator==4.4.2
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
Collecting decorator==4.4.2
  Downloading https://pypi.tuna.tsinghua.edu.cn/packages/ed/1b/72a1821152d07cf1d8b6fce298aeb06a7eb90f4d6d41acec9861e7cc6df0/decorat
or-4.4.2-py2.py3-none-any.whl (9.2 kB)
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Installing collected packages: decorator
  Attempting uninstall: decorator
    WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
    Found existing installation: decorator 5.0.6
    Uninstalling decorator-5.0.6:
      Successfully uninstalled decorator-5.0.6
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Successfully installed decorator-4.4.2
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)

D:\Program Files\cmder_mini
$ python -m pip install paddlepaddle==2.1.0 -f https://www.paddlepaddle.org.cn/whl/openblas/stable/noavx.html --no-index -i https:/
/mirror.baidu.com/pypi/simple
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Looking in links: https://www.paddlepaddle.org.cn/whl/openblas/stable/noavx.html
Collecting paddlepaddle==2.1.0
  Using cached https://paddle-wheel.bj.bcebos.com/2.1.0win/win-whl/open/noavx/paddlepaddle-2.1.0-cp38-cp38-win_amd64.whl (26.4 MB)
Requirement already satisfied: requests>=2.20.0 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (2.25.1)
Requirement already satisfied: astor in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (0.8.1)
Requirement already satisfied: Pillow in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (8.2.0)
Requirement already satisfied: decorator==4.4.2 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (4.4.2)
Requirement already satisfied: protobuf>=3.1.0 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (3.15.8)
Requirement already satisfied: numpy<=1.19.3,>=1.13 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (1.19.3
)
Requirement already satisfied: six in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (1.15.0)
Requirement already satisfied: gast==0.3.3 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (0.3.3)
Requirement already satisfied: idna<3,>=2.5 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle==2.1.
0) (2.10)
Requirement already satisfied: certifi>=2017.4.17 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle
==2.1.0) (2020.12.5)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepad
dle==2.1.0) (1.26.4)
Requirement already satisfied: chardet<5,>=3.0.2 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle=
=2.1.0) (4.0.0)
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Installing collected packages: paddlepaddle
  Attempting uninstall: paddlepaddle
    WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
    Found existing installation: paddlepaddle 2.0.2
    Uninstalling paddlepaddle-2.0.2:
      Successfully uninstalled paddlepaddle-2.0.2
WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
Successfully installed paddlepaddle-2.1.0

D:\Program Files\cmder_mini
$ python
Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import paddle
WARNING: AVX is not support on your machine. Hence, no_avx core will be imported, It has much worse preformance than avx core.
D:\Program Files\python\lib\site-packages\setuptools\depends.py:2: DeprecationWarning: the imp module is deprecated in favour of im
portlib; see the module's documentation for alternative uses
  import imp
>>> paddle.utils.run_check()
Running verify PaddlePaddle program ...
PaddlePaddle works well on 1 CPU.
W0701 17:32:02.829372  4732 fuse_all_reduce_op_pass.cc:76] Find all_reduce operators: 2. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 2.
PaddlePaddle works well on 2 CPUs.
PaddlePaddle is installed successfully! Let's start deep learning with PaddlePaddle now.

**Comment by wawltor** [comment]
> OK 安装成功了。
> D:\Program Files\cmder_mini
> $ pip install decorator==4.4.2
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Looking in indexes: https://pypi.tuna.tsinghua.edu.cn/simple
> Collecting decorator==4.4.2
> Downloading https://pypi.tuna.tsinghua.edu.cn/packages/ed/1b/72a1821152d07cf1d8b6fce298aeb06a7eb90f4d6d41acec9861e7cc6df0/decorat
> or-4.4.2-py2.py3-none-any.whl (9.2 kB)
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Installing collected packages: decorator
> Attempting uninstall: decorator
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Found existing installation: decorator 5.0.6
> Uninstalling decorator-5.0.6:
> Successfully uninstalled decorator-5.0.6
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Successfully installed decorator-4.4.2
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> 
> D:\Program Files\cmder_mini
> $ python -m pip install paddlepaddle==2.1.0 -f https://www.paddlepaddle.org.cn/whl/openblas/stable/noavx.html --no-index -i https:/
> /mirror.baidu.com/pypi/simple
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Looking in links: https://www.paddlepaddle.org.cn/whl/openblas/stable/noavx.html
> Collecting paddlepaddle==2.1.0
> Using cached https://paddle-wheel.bj.bcebos.com/2.1.0win/win-whl/open/noavx/paddlepaddle-2.1.0-cp38-cp38-win_amd64.whl (26.4 MB)
> Requirement already satisfied: requests>=2.20.0 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (2.25.1)
> Requirement already satisfied: astor in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (0.8.1)
> Requirement already satisfied: Pillow in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (8.2.0)
> Requirement already satisfied: decorator==4.4.2 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (4.4.2)
> Requirement already satisfied: protobuf>=3.1.0 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (3.15.8)
> Requirement already satisfied: numpy<=1.19.3,>=1.13 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (1.19.3
> )
> Requirement already satisfied: six in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (1.15.0)
> Requirement already satisfied: gast==0.3.3 in d:\program files\python\lib\site-packages (from paddlepaddle==2.1.0) (0.3.3)
> Requirement already satisfied: idna<3,>=2.5 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle==2.1.
> 0) (2.10)
> Requirement already satisfied: certifi>=2017.4.17 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle
> ==2.1.0) (2020.12.5)
> Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepad
> dle==2.1.0) (1.26.4)
> Requirement already satisfied: chardet<5,>=3.0.2 in d:\program files\python\lib\site-packages (from requests>=2.20.0->paddlepaddle=
> =2.1.0) (4.0.0)
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Installing collected packages: paddlepaddle
> Attempting uninstall: paddlepaddle
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Found existing installation: paddlepaddle 2.0.2
> Uninstalling paddlepaddle-2.0.2:
> Successfully uninstalled paddlepaddle-2.0.2
> WARNING: Ignoring invalid distribution -addlepaddle (d:\program files\python\lib\site-packages)
> Successfully installed paddlepaddle-2.1.0
> 
> D:\Program Files\cmder_mini
> $ python
> Python 3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)] on win32
> Type "help", "copyright", "credits" or "license" for more information.
> 
> > > > import paddle
> > > > WARNING: AVX is not support on your machine. Hence, no_avx core will be imported, It has much worse preformance than avx core.
> > > > D:\Program Files\python\lib\site-packages\setuptools\depends.py:2: DeprecationWarning: the imp module is deprecated in favour of im
> > > > portlib; see the module's documentation for alternative uses
> > > > import imp
> > > > paddle.utils.run_check()
> > > > Running verify PaddlePaddle program ...
> > > > PaddlePaddle works well on 1 CPU.
> > > > W0701 17:32:02.829372  4732 fuse_all_reduce_op_pass.cc:76] Find all_reduce operators: 2. To make the speed faster, some all_reduce ops are fused during training, after fusion, the number of all_reduce ops is 2.
> > > > PaddlePaddle works well on 2 CPUs.
> > > > PaddlePaddle is installed successfully! Let's start deep learning with PaddlePaddle now.

OK  谢谢反馈问题！

**Comment by joytom0591** [comment]
貌似不支持python 3.10，频频报错。noAVX也仅支持python3.8以上。
